File: backend/app/schemas/generation.py
from datetime import datetime

from app.schemas.base import CamelModel


class ProposedEventRead(CamelModel):
    id: str
    routine_id: str
    title: str
    start: str
    end: str
    editable: bool


class GeneratedWeekRead(CamelModel):
    id: str
    week_start: str
    status: str
    events: list[ProposedEventRead]
    created_at: datetime | str


class GenerateRequest(CamelModel):
    routine_ids: list[str]
    week_start: str

File: frontend/src/types/generation.ts
// Week generation types

export type GenerationStatus = 'idle' | 'selecting' | 'chatting' | 'generating' | 'reviewing' | 'confirmed'

export interface ProposedEvent {
  id: string
  routineId: string
  title: string
  start: string
  end: string
  editable: boolean
}

export interface GeneratedWeek {
  id: string
  weekStart: string // ISO date (Monday)
  status: GenerationStatus
  events: ProposedEvent[]
  createdAt: string
}
